dotnet/src/webdriver/DevTools/IDevToolsSession.cs

```diff
@@ -40,6 +40,13 @@ public interface IDevToolsSession : IDisposable
         /// </summary>
         event EventHandler<DevToolsEventReceivedEventArgs> DevToolsEventReceived;
 
+        /// <summary>
+        /// Enables waiting for the debugger when opening a new target (tab for instance).
+        /// The target will be halted until the Runtime.runIfWaitingForDebugger command is invoked.
+        /// </summary>
+        /// <returns></returns>
+        Task EnableWaitForDebuggerOnStart();
```

Comment:
Running this way will infinitely increase interface. Today it is `EnableSomething()`, tomorrow - `DisableSomething()`, the day after tomorrow `EnableSomethingElse()`. I think of that probably it would be better to introduce some kind of `DevToolsOptions`. This object represents desired options in the given context, optionally provided by user in class constructor, or in method call.

Example (pseudo-code):
```csharp
var session = new DevToolsSession(..., new DevToolsOptions() { WaitForDebugger = true });
```

This above is just an example how we can introduce new API without breaking changes in future, `Options` can be extended. I am still not sure it is the best suggestion, need to dive deeper into issue.

PS: The biggest issue is that we cannot cover all scenarios provided by CDP in "simplified" API.

Reply:
Agreed, using an options class seems like a better approached. I've committed some changes to remove EnableWaitForDebuggerOnStart function from the IDevToolsSession interface. Introduced DevToolsSessionOptions.cs to enable this functionality. By default it will be turned off.

Reply:
Looks better, thanks. Seems `protocolVersion` argument can be a part of new Options class. If you can, please make these contructors/methos as obsolete and point users to use new Options class.

Reply:
Committed some changes to make the DevToolsOptions more prominent in the API. Kept the old version and marked them as obsolete to prevent introducing issues for consumers who these specific functions. 

Setting the AutoDetectDevToolsProtocolVersion option is moved downstream (DevToolsSession.cs) instead of on the specific Driver implementations.

Reply:
Thanks, looks good.